docs/design-docs/overview.md

```diff
@@ -0,0 +1,144 @@
+6.	Provide a unified interface for all benchmarks, including how to run on different device vendor (NVIDIA/AMD), how to run on different mode (local/mpi), how to pass configurations and save results.
+
+## Design Principle
+
+1.	Make the tool a standard python package in pythonic way.
+2.	Only add necessary new features.
+3.	Keep frequently used benchmarks at the first stage.
+4.	Refactor code structure and internal interfaces, more features/benchmarks to be added in the future.
+
+## Architecture
+
+![SuperBench Workflow](../assets/executor_workflow.png)
+
+## Pipeline
+
+![Pipeline](../assets/executor-pipeline.png)
+
+1.	User prepares config file and host file. Both files could be omitted and use the default config or command line arguments to specify.
+2.	User runs sb cli on head node. Command line interface could accept a set of arguments and provide help information to user.
+3.	Sb cli parses the input config file, host file, and arguments, loads into one config object, and calls sb runner to start a run.
+4.	Sb runner parses the config object, and for all compute nodes in config,
+    1. Check the connection.
+    2. Check docker environment and sb Docker image.
+    3. Start an sb Docker container, mount necessary paths.
+    4. Prepare running context, including local package code, config file, SSH key pairs, SSH config for password less use. Tar all context and distribute to compute node.
+    5. Prepare output path.
+    6. Start SSH service and check inter-connections.
+5.	Sb runner loops all benchmarks and all modes in each benchmark. For each mode in each benchmark, sb runner calls sb executor inside Docker container on corresponding compute nodes to start an execution.
+6.	Sb executor parses the config object, start one benchmark inside Docker container.
+7.	Sb executor gets return code and results once the benchmark finished.
+8.	Sb executor sends the return code and results back to sb runner. Sb runner checks return code and moves to the next execution.
+9.	Once finished, sb runner reduces results on all compute nodes, save log and results files, then summarize running results to sb cli.
+10.	Sb cli returns what it gets to user.
+
+## Components
+
+### SB CLI
+
+SB cli is a command line interface for end users to use SuperBench tool and run benchmarks. The cli should provide a set of commands and corresponding help information to users. Details will be discussed in [interfaces section](#interfaces).
+
+### SB Runner
+
+SB runner is the component to configure environments, prepare context, run benchmarks, and collect log and results on all compute nodes. It will control the running logic, including when to start, which node or a set of nodes to run, whether a barrier is needed, etc. Sb runner will either communicate with host through SSH to configure running environments and prepare context, or communicate with sb executor inside Docker container to execute benchmarks and collect log and results on each node.
```

Comment:
`SuperBenchRunner` is the component to configure environments, prepare context, run benchmarks, collect log from nodes, and summarize results. It controls the running logic, including when to start, which node or a set of nodes to run, whether a barrier is needed, etc. `SuperBenchRunner` either communicates with host through SSH to configure running environments and prepare context, or talk with `SuperBenchExecutor` inside Docker container to execute benchmarks and collect log and results on each node.